slicedimage/io.py

```diff
@@ -26,17 +26,14 @@ def infer_backend(baseurl, allow_caching=True):
 
     if parsed.scheme in ("http", "https"):
         backend = HttpBackend(baseurl)
+        if allow_caching:
+            backend = CachingBackend("~/tmp", backend)
```

Comment:
As far as I know, this does not do path expansion.  I think you want:

`backend = CachingBackend(os.path.expanduser("~/.starfish-cache"), backend)`